compiler/noirc_frontend/src/ast/visitor.rs

```diff
@@ -1339,73 +1342,84 @@ impl UnresolvedType {
+                if visitor.visit_mutable_reference_type(unresolved_type, self.location.span) {
                     unresolved_type.accept(visitor);
                 }
             }
             UnresolvedTypeData::Tuple(unresolved_types) => {
-                if visitor.visit_tuple_type(unresolved_types, self.span) {
+                if visitor.visit_tuple_type(unresolved_types, self.location.span) {
                     visit_unresolved_types(unresolved_types, visitor);
                 }
             }
             UnresolvedTypeData::Function(args, ret, env, unconstrained) => {
-                if visitor.visit_function_type(args, ret, env, *unconstrained, self.span) {
+                if visitor.visit_function_type(args, ret, env, *unconstrained, self.location.span) {
                     visit_unresolved_types(args, visitor);
                     ret.accept(visitor);
                     env.accept(visitor);
                 }
             }
             UnresolvedTypeData::AsTraitPath(as_trait_path) => {
-                if visitor.visit_as_trait_path_type(as_trait_path, self.span) {
-                    as_trait_path.accept(self.span, visitor);
+                if visitor.visit_as_trait_path_type(as_trait_path, self.location.span) {
+                    as_trait_path.accept(self.location.span, visitor);
                 }
             }
-            UnresolvedTypeData::Expression(expr) => visitor.visit_expression_type(expr, self.span),
+            UnresolvedTypeData::Expression(expr) => {
+                visitor.visit_expression_type(expr, self.location.span);
+            }
             UnresolvedTypeData::FormatString(expr, typ) => {
-                if visitor.visit_format_string_type(expr, typ, self.span) {
+                if visitor.visit_format_string_type(expr, typ, self.location.span) {
                     typ.accept(visitor);
                 }
             }
-            UnresolvedTypeData::String(expr) => visitor.visit_string_type(expr, self.span),
-            UnresolvedTypeData::Unspecified => visitor.visit_unspecified_type(self.span),
-            UnresolvedTypeData::Quoted(typ) => visitor.visit_quoted_type(typ, self.span),
-            UnresolvedTypeData::FieldElement => visitor.visit_field_element_type(self.span),
+            UnresolvedTypeData::String(expr) => visitor.visit_string_type(expr, self.location.span),
+            UnresolvedTypeData::Unspecified => visitor.visit_unspecified_type(self.location.span),
+            UnresolvedTypeData::Quoted(typ) => visitor.visit_quoted_type(typ, self.location.span),
+            UnresolvedTypeData::FieldElement => {
+                visitor.visit_field_element_type(self.location.span);
+            }
             UnresolvedTypeData::Integer(signdness, size) => {
-                visitor.visit_integer_type(*signdness, *size, self.span);
+                visitor.visit_integer_type(*signdness, *size, self.location.span);
```

Comment:
typo: `signdness` -> `signedness`
```suggestion
                visitor.visit_integer_type(*signedness, *size, self.location.span);
```

Reply:
Done, thanks!